plugins/ui-update/notifications.ts

```diff
@@ -0,0 +1,186 @@
+import { i18nMark } from "@lingui/react";
+import * as semver from "semver";
+import { filter } from "rxjs/operators";
+import {
+  ToastAppearance,
+  ToastCallbackType,
+  ToastNotification
+} from "@extension-kid/toast-notifications";
+
+import { getAction$, getUiMetadata$ } from "./streams";
+import { UIActions, UIActionType } from "./types/UIAction";
+import { rollbackUI, updateUI } from "./commands";
+import { getNotificationService } from "./utils";
+
+const LOCAL_DEV_VERSION = "0.0.0";
+const activeNotifications: string[] = [];
+
+function clearActiveNotifications() {
+  if (activeNotifications.length) {
+    activeNotifications.splice(0, activeNotifications.length);
+  }
+}
+
+function removeActiveNotification(id: string) {
+  const notificationIndex = activeNotifications.indexOf(id);
+  if (notificationIndex !== -1) {
+    activeNotifications.splice(notificationIndex, 1);
+  }
+}
+
+function displayNotification(notification: ToastNotification) {
+  if (!activeNotifications.includes(notification.id)) {
+    const ns = getNotificationService();
```

Comment:
What if we don't have the service on the container? :)

Reply:
This is no longer an issue with the latest refactor.

Reply:
Nit-pick: can we not abbreviate the `NotificationService` with `ns`.

Also we could also just do `getNotificationService().push(notification)` instead right?